### ISSUE TITLE
generator: Trailing underscore in test method name.

### ISSUE THREAD

**Issue by Insti** [exercism/ruby#668]
If the [canonical data description of a problem](https://github.com/exercism/x-common/blob/51f1341e4dfd070535e4d0e0039652a98cea7dc8/exercises/collatz-conjecture/canonical-data.json#L30) has trailing whitespace.
```
"description": "negative value is an error ",
```

The [xruby generated test](https://github.com/exercism/xruby/pull/667/files#diff-05abd772bcd323162673971a2a205fd6R27) will have trailing underscore.
```
def test_negative_value_is_an_error_
```

Is the correct thing to do: 
a) Fix the canonical data?
b) Fix the xruby generator?
c) Do both?
d) Do something else?

**Comment by kotp** [comment]
We can't "control" the canonical, so defend against it here.  If it is corrected and never shows up in canonical data again, it will be fine.

But definitely mention it or correct it in canonical.

**Comment by Insti** [comment]
canonical data fixed by https://github.com/exercism/problem-specifications/pull/839

**Comment by kotp** [comment]
So now to defend against it here, I think?

**Comment by Insti** [comment]
Resolved by #732 which ensures test names won't have extra underscores if the canonical data has trailing (or leading) whitespace.
